bert_reranker/data/convert_faq_json.py

```diff
@@ -6,49 +6,89 @@
 logger = logging.getLogger(__name__)
 
 
-def make_qa_pairs_faq(faq_path, n_wrong_answers=2, seed=42):
-    with open(faq_path, "r", encoding="utf-8") as fh:
-        faq = json.load(fh)
+def make_qa_pairs_faq(json_file, n_wrong_answers, seed):
 
     random.seed(seed)
-    all_questions = []
+    all_questions_to_answer_index = {}
     all_answers = []
 
-    for k, v in faq.items():
+    answer_size = None
+
+    for k, v in json_file.items():
         if k != "document_URL":
-            all_questions.append(k)
-            all_answers.append("".join(faq[k]["plaintext"]))
+            answer = "".join(json_file[k]["plaintext"])
+
+            if answer not in all_answers:
+                all_answers.append(answer)
+            answer_index = all_answers.index(answer)
+            all_questions_to_answer_index[k] = answer_index
 
     qa_pairs = []
-    for idx, question in enumerate(all_questions):
-        correct_answer = all_answers[idx]
+    for question, answer_index in all_questions_to_answer_index.items():
+        correct_answer = all_answers[answer_index]
         wrong_answers = all_answers.copy()
         wrong_answers.remove(correct_answer)
         random.shuffle(wrong_answers)
 
         candidate_answers = []
         candidate_answers.append(correct_answer)
-        candidate_answers.extend(wrong_answers[:n_wrong_answers])
+        if n_wrong_answers > 0:
+            wrong_answers = wrong_answers[:n_wrong_answers]
+
+        if answer_size is None:
+            answer_size = len(wrong_answers)
+
+        if answer_size != len(wrong_answers):
+            raise ValueError('different number of answers')
+
+        candidate_answers.extend(wrong_answers)
         qa_pairs.append([question, candidate_answers])
 
+    logger.info('every question has {} wrong 1 answers'.format(answer_size))
```

Comment:
should this read `every question has {} wrong answers` ?

Reply:
indeed :)

Reply:
fixed.